Providers/PocketFeed.swift

```diff
@@ -0,0 +1,134 @@
+                let imageURLS = storyDict["image_src"] as? String,
+                let title = storyDict["title"] as? String,
+                let description = storyDict["excerpt"] as? String else {
+                    return nil
+            }
+            guard let url = URL(string: urlS), let imageURL = URL(string: imageURLS) else {
+                return nil
+            }
+            return PocketStory(url: url, title: title, storyDescription: description, imageURL: imageURL, domain: domain)
+        })
+    }
+}
+
+private class PocketError: MaybeErrorType {
+    var description = "Failed to load from API"
+}
+
+class Pocket {
+
+    let pocketGlobalFeed: String
+    // Allow endPoint to be overriden for testing
+    init(endPoint: String? = nil) {
+        pocketGlobalFeed = endPoint ?? PocketGlobalFeed
+    }
+
+    lazy fileprivate var alamofire: SessionManager = {
+        let ua = UserAgent.fxaUserAgent //TODO: use a different UA
+        let configuration = URLSessionConfiguration.default
+        return SessionManager.managerWithUserAgent(ua, configuration: configuration)
+    }()
+
+    func findCachedResposneFor(request: URLRequest) -> [String: Any]? {
+        let cachedResponse = URLCache.shared.cachedResponse(for: request)
+        guard let cachedAtTime = cachedResponse?.userInfo?["cache-time"] as? Timestamp, (Date.now() - cachedAtTime) < MaxCacheAge else  {
+            return nil
+        }
+
+        guard let data = cachedResponse?.data, let json = try? JSONSerialization.jsonObject(with: data, options: []) else {
+            return nil
+        }
+
+        return json as? [String: Any]
+    }
+
+    func cacheResponseFor(response: HTTPURLResponse?, request: URLRequest, data: Data?) {
+        guard let resp = response, let data  = data else {
+            return
+        }
+        let metadata = ["cache-time": Date.now()]
+        let cachedResp = CachedURLResponse(response: resp, data: data, userInfo: metadata, storagePolicy: .allowed)
+        URLCache.shared.removeCachedResponse(for: request)
+        URLCache.shared.storeCachedResponse(cachedResp, for: request)
+    }
+
+    // Fetch items from the global pocket feed
+    func globalFeed(items: Int = 2) -> Deferred<Maybe<Array<PocketStory>>> {
+        let deferred = Deferred<Maybe<Array<PocketStory>>>()
+
+        guard let request = createGlobalFeedRequest(items: items) else {
+            return deferMaybe(PocketError())
+        }
+
+        if let cachedResponse = findCachedResposneFor(request: request), let items = cachedResponse["list"] as? Array<[String: Any]> {
+            deferred.fill(Maybe(success: PocketStory.parseJSON(list: items)))
+            return deferred
+        }
+
+        alamofire.request(request).validate(contentType: ["application/json"]).responseJSON { response in
+            guard response.error == nil, let result = response.result.value as? [String: Any] else {
```

Comment:
Elsewhere we use SwiftyJSON, is there a particular JSON lib usage we follow ?
NSDictionary to array casting (particularly on main thread) has been a perf problem for me in apps. I haven't used SwiftyJSON but I was planning to read the code to see if they maintain the JSON as an NSDictionary.
Ok, I see later in the patch this is bounded to small list size, so I think this is a non-issue here.